content/en/blog/2024/prom-and-otel/index.md

```diff
@@ -0,0 +1,671 @@
+
+The Prometheus exporter allows you to ship data in the Prometheus format, which
+is then scraped by a Prometheus server. It's used to report metrics via the
+Prometheus scrape HTTP endpoint. You can learn more by trying out this
+[example](https://github.com/open-telemetry/opentelemetry-go/tree/main/example/prometheus).
+However, the scraping won't really scale, as all the metrics are sent in a
+single scrape.
+
+To get around the scaling concern, you can alternatively use the Prometheus
+Remote Write exporter, which allows you to push data to Prometheus from multiple
+Collector instances with no issues. Since Prometheus also accepts remote write
+ingestion, you can also use this exporter if you are generating OTel metrics and
+want to ship them to a backend that is compatible with Prometheus remote write.
+To learn more about the architecture of both exporters, see
+[Use Prometheus Remote Write exporter](https://grafana.com/blog/2023/07/20/a-practical-guide-to-data-collection-with-opentelemetry-and-prometheus/#6-use-prometheus-remote-write-exporter).
+
+## Using the Target Allocator
+
+Scalability is a common challenge with Prometheus; that's the ability to
+effectively maintain performance and resource allocation while managing an
+increasing number of monitored targets and metrics. One option to help with this
+is sharding the workload based on labels or dimensions,
+[which means using multiple Prometheus instances to handle your metrics according to specific parameters](https://medium.com/wish-engineering/horizontally-scaling-prometheus-at-wish-ea4b694318dd).
+This could help decrease the burden on individual instances. However, there are
+two things to consider with this approach.
+
+The first is that to get around querying sharded instances, you need a
+management instance; this means that you need to have N+1 Prometheus instances,
+where the +1’s memory is equal to N, thereby doubling your memory requests.
+Secondly, Prometheus sharding requires that each instance scrape the target,
+even if it’s going to be dropped.
+
+Something to note is that if you can have a Prometheus instance with the
+combined amount of memory of individual instances, there is not much benefit to
+sharding, since you can scrape everything directly using the larger instance. A
+reason that people shard is usually for some amount of fault tolerance. For
+example, if one Prometheus instance is out of memory (OOM), then your entire
+alerting pipeline won't be offline.
+
+Luckily, the OTel Operator’s Target Allocator (TA) is able to help with some of
+this. For instance, it can automatically drop any targets it knows won’t be
+scraped. Whereas if you shard with `hashmod`, you'll need to
+[update your config based on the number of replicas you have](https://www.robustperception.io/scaling-and-federating-prometheus/).
+Plus, if you’re already collecting Prometheus metrics about your Kubernetes
+infrastructure, using the TA is a great option.
```

Comment:
```suggestion
Luckily, the OTel Operator’s Target Allocator (TA) is able to help with some of
this. For instance, it can automatically drop any targets it knows won’t be
scraped. The TA also automatically shards the targets for you, whereas if you shard with `hashmod` you'll need to
[update your config based on the number of replicas you have](https://www.robustperception.io/scaling-and-federating-prometheus/). The TA also allows you to continue using resources like PodMonitor and ServiceMonitor to continue collecting Prometheus metrics about your Kubernetes
infrastructure.
```